src/package-managers/bun.ts

```diff
@@ -0,0 +1,42 @@
+import spawn from 'spawn-please'
+import { Index } from '../types/IndexType'
+import { NpmOptions } from '../types/NpmOptions'
+import { latest as npmLatest, list as npmList } from './npm'
+
+/**
+ * (Bun) Fetches the list of all installed packages.
+ */
+export const list = npmList
+
+/**
+ * (Bun) Fetches the latest version of a package.
+ */
+export const latest = npmLatest
+
+/**
+ * Spawn bun.
+ *
+ * @param args
+ * @param [bunOptions={}]
+ * @param [spawnOptions={}]
+ * @returns
+ */
+async function spawnBun(
+  args: string | string[],
+  bunOptions: NpmOptions = {},
+  spawnOptions: Index<any> = {},
+): Promise<any> {
+  // Bun not yet supported on Windows.
+  // @see https://github.com/oven-sh/bun/issues/43
+  if (process.platform === 'win32') {
+    throw new Error('Bun not yet supported on Windows')
```

Comment:
Please use https://github.com/raineorshine/npm-check-updates/blob/main/src/lib/programError.ts, which has different error behavior for CLI mode and programmatic mode.